wrapper/CSharp/wolfTPM.cs

```diff
@@ -0,0 +1,839 @@
+        PrintByteArray(pub_buffer);
+        PrintByteArray(priv_buffer);
+
+        dev = wolfTPM2_New();
+        if (dev == IntPtr.Zero) {
+            Console.WriteLine("wolfTPM2_New() failed.");
+            ret = -1;
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            priv_key = wolfTPM2_GetNewKey();
+            if (priv_key == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetNewKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            parent_key = wolfTPM2_GetNewKey();
+            if (parent_key  == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetNewKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            blob = wolfTPM2_GetNewKeyBlob();
+            if (blob == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetNewKeyBlob() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_ReadPublicKey(dev, parent_key, TPM2_PARENT_HANDLE);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_ReadPublicKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_SetKeyAuthPassword(parent_key, pwBytes,
+                                              pwBytes.Length);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_SetKeyAuthPassword() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_ImportRsaPrivateKey(dev, parent_key, blob,
+                                               pub_buffer, pub_buffer.Length,
+                                               exp, priv_buffer,
+                                               priv_buffer.Length,
+                                               TPM_ALG_NULL, TPM_ALG_NULL);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_ImportRsaPrivateKey() failed.");
+            }
+        }
+
+        if (priv_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(priv_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey(priv_key) failed.");
+            }
+        }
+
+        if (parent_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(parent_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey(parent_key) failed.");
+            }
+        }
+
+        if (dev != IntPtr.Zero) {
+            if (wolfTPM2_Free(dev) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_Free() failed.");
+            }
+        }
+
+        Console.WriteLine("Leaving TryImportRSAPrivateKey() with ret=" + ret);
+        Console.WriteLine("");
+    }
+
+    public static void Main(string[] args) {
```

Comment:
I agree with @elms . This file should produce a library, not an app. Please move the `Main` and "Try" functions to a new .cs and update the build steps. Use `libwolfTPM.cs` and `wolfTPM.cs`.